platform/android/app/src/test/kotlin/coop/polypoly/polypod/polyout/PolyOutTest.kt

```diff
@@ -0,0 +1,298 @@
+
+        companion object {
+            private val wrapped = KeyStore.getInstance("BKS", "BC")
+            internal val storedKeys = mutableMapOf<String, KeyStore.Entry>()
+        }
+    }
+
+    class FakeAesKeyGenerator : KeyGeneratorSpi() {
+        private val wrapped = KeyGenerator.getInstance("AES", "BC")
+        private var lastSpec: AlgorithmParameterSpec? = null
+
+        override fun engineInit(random: SecureRandom?) = wrapped.init(random)
+
+        override fun engineInit(
+            params: AlgorithmParameterSpec?,
+            random: SecureRandom?
+        ) = wrapped.init(random).also {
+            lastSpec = params
+        }
+
+        override fun engineInit(
+            keysize: Int,
+            random: SecureRandom?
+        ) = wrapped.init(keysize, random)
+
+        override fun engineGenerateKey(): SecretKey = wrapped.generateKey()
+    }
+
+    private var polyOut: PolyOut = PolyOut(
+        context = InstrumentationRegistry.getInstrumentation().targetContext
+    )
+    private var context = InstrumentationRegistry
+        .getInstrumentation()
+        .targetContext
+    private var resolver = Shadows.shadowOf(context.contentResolver)
+
+    @Before
+    fun setup() {
+        Security.addProvider(object : Provider("AndroidKeyStore", 1.0, "") {
+            init {
+                put("KeyStore.AndroidKeyStore", FakeKeyStore::class.java.name)
+                put("KeyGenerator.AES", FakeAesKeyGenerator::class.java.name)
+            }
+        })
+
+        FeatureStorage.activeFeature = MockFeature(
+            id = "testFeatureId",
+            name = "testFeature",
+            author = "test",
+            description = "",
+            primaryColor = 0,
+            thumbnailColor = 0,
+            thumbnail = null
+        )
+    }
+
+    @After
+    fun teardown() {
+        //
+    }
+
+    @Test
+    fun importOneArchiveWorks() {
+        val url = InstrumentationRegistry
+            .getInstrumentation()
+            .javaClass.classLoader!!
+            .getResource("testZip.zip")
+            .toString()
+        val inputStream = InstrumentationRegistry
+            .getInstrumentation()
+            .javaClass.classLoader!!
+            .getResource("testZip.zip")
+            .openStream()
+        resolver.registerInputStream(Uri.parse(url), inputStream)
+
+        val zipId = runBlocking {
+            polyOut.importArchive(url = url)
+        }
+
+        Truth.assertThat(zipId).isNotEmpty()
+
+        val files = runBlocking {
+            polyOut.readDir(zipId!!)
+        }
+
+        Truth.assertThat(files).isNotEmpty()
+        val hasFile = files.any {
+            it["path"] == "testZip/testfile.rtf"
+        }
+        Truth.assertThat(hasFile).isTrue()
+    }
+
+    @Test
+    fun importMultipleArchivesWorks() {
+        val url1 = InstrumentationRegistry
+            .getInstrumentation()
+            .javaClass.classLoader!!
+            .getResource("multipleZips1.zip")
```

Comment:
Same as said above, extract up to here to a variable and reuse it in the assignments below.